File: strings/1277-short-attendance.go
package main

import (
	"fmt"
	"strings"
)

type student struct {
	name      string
	frequency string
}

func main() {
	var t int
	var n int
	fmt.Scan(&t)
	for t != 0 {
		fmt.Scan(&n)
		students := make([]student, n)
		for i := 0; i < n; i++ {
			fmt.Scan(&students[i].name)
		}
		for i := 0; i < n; i++ {
			fmt.Scan(&students[i].frequency)
		}
		first := true
		for _, s := range students {
			medical := strings.Count(s.frequency, "M")
			present := float64(strings.Count(s.frequency, "P"))
			total := float64(len(s.frequency) - medical)
			if (present / total) < float64(0.75) {
				if !first {
					fmt.Printf(" ")
				}
				first = false
				fmt.Printf(s.name)
			}
		}
		fmt.Println()
		t--
	}
}

File: strings/1277-short-attendance.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<string> attendance;
    vector<string> students;
    string input;
    bool first;
    int n;
    int m;

    cin >> n;
    while (n > 0) {
        cin >> m;
        attendance.clear();
        students.clear();
        for (int i = 0; i < m; i++) {
            cin >> input;
            students.push_back(input);
        }
        for (int i = 0; i < m; i++) {
            cin >> input;
            attendance.push_back(input);
        }
        first = true;
        for (int i = 0; i < m; i++) {
            int medical = count(attendance[i].begin(), attendance[i].end(), 'M');
            int present = count(attendance[i].begin(), attendance[i].end(), 'P');
            float total = attendance[i].size() - medical;
            if ((present / total) < 0.75) {
                if (!first) {
                    cout << " ";
                }
                cout << students[i];
                first = false;
            }
        }
        cout << endl;
        n--;
    }

    return 0;
}
